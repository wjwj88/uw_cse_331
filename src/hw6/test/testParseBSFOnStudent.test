# test parser and BSF on student graph  generic version-Jun Wang 5:00

# build graph
LoadGraph student student.tsv

# check graph is built correctly
ListNodes student 
ListChildren student Amanda
ListChildren student Jack
ListChildren student Bill

# check BFS
FindPath student Amanda Amanda
FindPath student Tom Mike
FindPath student Amanda Jack