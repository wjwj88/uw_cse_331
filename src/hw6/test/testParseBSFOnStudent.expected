# test parser and BSF on student graph  generic version-Jun Wang 5:00

# build graph
loaded graph student 

# check graph is built correctly
student contains: Amanda Bill Jack Jun Mike Tom
the children of Amanda in student are: 
the children of Jack in student are: Bill(US) Tom(US)
the children of Bill in student are: Jack(US) Jun(China) Mike(China) Tom(US)

# check BFS
path from Amanda to Amanda:
path from Tom to Mike: 
Tom to Bill via US
Bill to Mike via China  
path from Amanda to Jack: 
no path found  